velox/common/memory/Memory.cpp

```diff
@@ -181,6 +181,11 @@ bool MemoryManager::growPool(MemoryPool* pool, uint64_t incrementBytes) {
   return arbitrator_->growMemory(pool, candidates, incrementBytes);
 }
 
+uint64_t MemoryManager::shrinkPools(uint64_t targetBytes) {
+  VELOX_CHECK_NOT_NULL(arbitrator_);
+  return arbitrator_->shrinkMemory(getAlivePools(), targetBytes);
```

Comment:
getAlivePools() holds the memory pool while shrinking memory. So let's change to the following (as what growPool does)
```
  // Holds a shared reference to each alive memory pool in 'pools_' to keep
  // their aliveness during the memory arbitration process.
  std::vector<std::shared_ptr<MemoryPool>> candidates = getAlivePools();
  return arbitrator_->shrinkMemory(candidates, targetBytes);
```

Reply:
What's the difference? Calling getAlivePools() will pass a copy of every shared_ptr to shrinkMemory() and hence keeping them alive.

Reply:
You are right. Could you make a change to growMemory? Thanks!